include/dfu/dfu_target.rst

```diff
@@ -0,0 +1,48 @@
+.. _lib_dfu_target:
+
+DFU target
+##########
+
+The DFU target library provides a common API for different types of firmware upgrades, for example, an MCUboot style upgrade or a modem firmware upgrade.
+Use this library in your component to do different types of firmware upgrades against a single interface.
+
+When initializing the DFU target library, you must provide information about the type of firmware upgrade.
+To do so automatically, send the first fragment of the firmware to the function :cpp:func:`dfu_target_img_type`.
+This function can identify all supported firmware upgrade types.
+The result of this call can then be given as input to the :cpp:func:`dfu_target_init` function.
+
+
+Supported DFU targets
+*********************
+
+Every supported DFU target must implement the set of functions defined in :file:`subsys/dfu/src/dfu_target.c`.
+
+The following sections describe the DFU targets that are currently supported.
+
+MCUboot style upgrades
+======================
+
+This type of firmware upgrade writes the data given to the :cpp:func:`dfu_target_write` function into the secondary slot specified by MCUboot's flash partitions.
+
+When the complete transfer is done, call the :cpp:func:`dfu_target_done` function to mark the firmware that is stored in the secondary slot as ready to be booted.
+On the next reboot, the device will run the new firmware.
+
+Modem firmware upgrades
+=======================
+
+This type of firmware upgrade opens a socket into the modem and passes the data given to the :cpp:func:`dfu_target_write` function through the socket.
+The modem stores the data in the memory location for firmware patches.
+If there is already a firmware patch stored in the modem, the library deletes the old firmware patch to make space for the new patch.
+
+When the complete transfer is done, call the :cpp:func:`dfu_target_done` function to request the modem to apply the patch, and to close the socket.
+On the next reboot, the modem will to try to apply the patch.
+
```

Comment:
Are there any important configuration options that should be mentioned? Looks like all targets are enabled by default, so there's no need?

Reply:
Thanks - I added another commit with this information.